tensorboard/webapp/plugins/npmi/views/data_selection/metric_search/metric_search_container.ts

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, ChangeDetectionStrategy} from '@angular/core';
+import {select, Store} from '@ngrx/store';
+
+import {map} from 'rxjs/operators';
+import {combineLatest} from 'rxjs';
+
+import {State} from '../../../../../app_state';
+import {
+  getMetricFilters,
+  getMetricsRegex,
+  getRunToMetrics,
+} from '../../../store';
+import {getRunSelection} from '../../../../../core/store/core_selectors';
+import * as npmiActions from '../../../actions';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+@Component({
+  selector: 'npmi-metric-search',
+  template: `
+    <metric-search-component
+      [regexFilterValue]="metricsRegex$ | async"
+      [completions]="completions$ | async"
+      [isRegexFilterValid]="isMetricsFilterValid$ | async"
+      (onRegexFilterValueChange)="onFilterChange($event)"
+      (onAddFilter)="onAddFilter($event)"
+    ></metric-search-component>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetricSearchContainer {
+  readonly metricsRegex$ = this.store.select(getMetricsRegex);
+  readonly activeRuns$ = this.store.pipe(select(getRunSelection)).pipe(
+    map((runs) => {
+      const activeRuns: string[] = [];
+      if (runs) {
+        for (const run of runs) {
+          if (run[1]) {
+            activeRuns.push(run[0]);
```

Comment:
I was a bit confused upon initial read.  `run` does not refer to a run, but rather the key-value pair `[run, isSelected]`.  Perhaps:
- rename `runs` -> `runSelection`
- and then
```
for (const [run, isSelected] of runSelection) {
  if (isSelected) { activeRuns.push(run); }
}
```
?

Reply:
Done.